src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDetectPreviewFeatureAnalyzer.cs

```diff
@@ -241,25 +179,8 @@ private static bool TryGetConstraintClauseNode(SyntaxList<TypeParameterConstrain
             foreach (SyntaxReference? syntaxReference in typeSymbolDeclaringReferences)
             {
                 SyntaxNode typeSymbolDefinition = syntaxReference.GetSyntax();
-                if (typeSymbolDefinition is ClassDeclarationSyntax classDeclaration)
-                {
-                    SeparatedSyntaxList<BaseTypeSyntax> baseListTypes = classDeclaration.BaseList.Types;
-                    if (TryGetPreviewInterfaceNodeForClassOrStructImplementingPreviewInterface(baseListTypes, previewInterfaceSymbol, out ret))
-                    {
-                        return ret;
-                    }
-                }
-                else if (typeSymbolDefinition is StructDeclarationSyntax structDeclaration)
-                {
-                    SeparatedSyntaxList<BaseTypeSyntax> baseListTypes = structDeclaration.BaseList.Types;
-                    if (TryGetPreviewInterfaceNodeForClassOrStructImplementingPreviewInterface(baseListTypes, previewInterfaceSymbol, out ret))
-                    {
-                        return ret;
-                    }
-                }
-                else if (typeSymbolDefinition is InterfaceDeclarationSyntax interfaceDeclaration)
+                if (typeSymbolDefinition is TypeDeclarationSyntax { BaseList.Types: var baseListTypes })
```

Comment:
This change fixes two issues:

- The null reference exception
- record declarations